addons/website_sale_delivery/controllers/main.py

```diff
@@ -34,7 +34,7 @@ def cart_carrier_rate_shipment(self, carrier_id, **kw):
         Monetary = request.env['ir.qweb.field.monetary']
         order = request.website.sale_get_order(force_create=True)
         res = {'carrier_id': carrier_id}
-        carrier = request.env['delivery.carrier'].browse(int(carrier_id))
+        carrier = request.env['delivery.carrier'].sudo().browse(int(carrier_id))
```

Comment:
@rdeodoo 

should probably check that delivery into _get_delivery_methods before to do the sudo

Reply:
As a security check? It is done